File: payloads/go/dns_lookup.go
package main

import (
	"fmt"
	"net"
)

func main() {
	addrs, err := net.LookupHost("google.com")
	if err != nil {
		panic(err)
	}
	fmt.Println("ESCAPED")
	fmt.Println(addrs)
}

File: payloads/c/dns_lookup.c
#include <stdio.h>
#include <netdb.h>

int main(void) {
    struct addrinfo *res = NULL;
    int err = getaddrinfo("google.com", "80", NULL, &res);
    if (err != 0) {
        fprintf(stderr, "dns lookup blocked: %s\n", gai_strerror(err));
        return 1;
    }
    freeaddrinfo(res);
    printf("ESCAPED\n");
    return 0;
}
